src/bin/cargo/commands/install.rs

```diff
@@ -172,6 +172,14 @@ pub fn exec(gctx: &mut GlobalContext, args: &ArgMatches) -> CliResult {
     let mut from_cwd = false;
 
     let source = if let Some(url) = args.get_one::<String>("git") {
+        if url.contains('#') {
+            return Err(anyhow::format_err!(
+                "invalid git url to install from\n\n\
+                    help: use `--rev <SHA>` to specify a commit"
+            )
+            .into());
+        }
+
         let url = url.into_url()?;
```

Comment:
Should this check be done on the parsed URL, rather than on the string?

Reply:
Thanks, yeah. That makes a lot of sense.